### PR TITLE
Elasticsearch auth implemented

### PR DESCRIPTION
**Pull request by SilverFire**
The code is tested and works as expected.
I'm not sure whether it can be covered by unit tests because it requires special ES server configuration. We can not mock it because CURL options are never exposed from the `httpRequest()`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ Yii Framework 2 elasticsearch extension Change Log
 - Bug #8: Fixed issue with running out of sockets when running a large number of requests by reusing curl handles (cebe)
 - Enh #28: AWS Elasticsearch service compatibility (andrey-bahrachev)
 - Enh #33: Implemented `Command::updateSettings()` and `Command::updateAnalyzers()` (githubjeka)
+- Enh: Implemented HTTP auth (silverfire)
```
Comment (samdark) [inline CHANGELOG.md]: `#50`

File: Connection.php [hunk 2]
```diff
@@ -35,19 +35,37 @@ class Connection extends Component
...
     /**
-     * @var array the active node. key of [[nodes]]. Will be randomly selected on [[open()]].
+     * @var string the active node. Key of one of the [[nodes]]. Will be randomly selected on [[open()]].
      */
     public $activeNode;
-    // TODO http://www.elastic.co/guide/en/elasticsearch/client/php-api/current/_configuration.html#_host_configuration
+
+    /**
+     * @var array|\Closure Authorization data used to connect to the ElasticSearch node.
+     * Possible array elements:
```
Comment (samdark) [inline Connection.php]: "Possible" contradicts "Array MUST contain both".

File: Connection.php [hunk 3]
```diff
@@ -35,19 +35,37 @@ class Connection extends Component
...
+     *  - `username`: the username for authentication.
+     *  - `password`: the password for authentication.
+     * Array MUST contain both `username` and `password`, when is set.
```
Comment (samdark) [inline Connection.php]: I guess it should be `Array either MUST contain both username or password on not contain any authentication credentials.`

File: Connection.php [hunk 4]
```diff
@@ -35,19 +35,37 @@ class Connection extends Component
...
     ];
     /**
-     * @var array the active node. key of [[nodes]]. Will be randomly selected on [[open()]].
+     * @var string the active node. Key of one of the [[nodes]]. Will be randomly selected on [[open()]].
      */
     public $activeNode;
-    // TODO http://www.elastic.co/guide/en/elasticsearch/client/php-api/current/_configuration.html#_host_configuration
+
+    /**
+     * @var array|\Closure Authorization data used to connect to the ElasticSearch node.
```
Comment (samdark) [inline Connection.php]: I guess you've meant authentication, not authorization.

**Comment by SilverFire** [comment]
Updated, thank you @samdark

**Comment by samdark** [comment]
Looks OK to me.

**Pull request merged**